src/components/TaskStatusBadge.tsx

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import Badge, { Variant } from "@leafygreen-ui/badge";
+import { TaskStatus } from "types/task";
+import styled from "@emotion/styled/macro";
+
+const mapTaskStatusToBadgeVariant = {
+  [TaskStatus.Inactive]: Variant.LightGray,
+  [TaskStatus.Unstarted]: Variant.LightGray,
+  [TaskStatus.Undispatched]: Variant.LightGray,
+  [TaskStatus.Started]: Variant.Yellow,
+  [TaskStatus.Dispatched]: Variant.Yellow,
+  [TaskStatus.Succeeded]: Variant.Green,
+  [TaskStatus.Failed]: Variant.Red,
+  [TaskStatus.StatusBlocked]: Variant.DarkGray,
+  [TaskStatus.StatusPending]: Variant.LightGray
+};
+
+const failureColors = {
+  text: "#800080",
+  border: "#CC99CC",
+  fill: "#E6CCE6"
+};
+
+// the status colors that are not supported by the leafygreen Badge variants
+const mapUnsupportedBadgeColors = {
+  [TaskStatus.SystemFailed]: failureColors,
+  [TaskStatus.TestTimedOut]: failureColors,
+  [TaskStatus.SetupFailed]: {
+    border: "#E7DBEC",
+    fill: "#F3EDF5",
+    text: "#877290"
+  }
+};
+
+interface BadgeColorProps {
+  border: string;
+  fill: string;
+  text: string;
+}
+
+// only use for statuses whose color is not supported by leafygreen badge variants, i.e. SystemFailed, TestTimedOut, SetupFailed
+const StyledBadge = styled(Badge)`
+  border-color: ${(props: BadgeColorProps) => props.border} !important;
+  background-color: ${(props: BadgeColorProps) => props.fill} !important;
+  color: ${(props: BadgeColorProps) => props.text} !important;
+`;
```

Comment:
if you're moving these somewhere to be shared two comonents then i would think it would be in a shared folder/file as opposed to another component.

Reply:
src/components is the "shared" folder according to fractal pattern

Reply:
ah, I think I know what you mean. I didnt export styled badge bc right now it's only used by TaskStatusBadge. TaskStatusBadge is the shared component.